### PR TITLE
Be able to parse more date string types

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Comment by vangheem** [comment]
tests running: http://jenkins.plone.org/job/pull-request-5.0/871/

File: plone/app/querystring/indexmodifiers/query_index_modifiers.py [hunk 1]
```diff
@@ -60,7 +60,10 @@ def __call__(self, value):
             try:
                 query = datetime.strptime(query, "%Y-%m-%d")
             except:
-                query = query.encode("utf-8")
+                try:
+                    query = datetime.strptime(query, "%m/%d/%Y")
+                except:
```
Comment (gforcada) [inline plone/app/querystring/indexmodifiers/query_index_modifiers.py]: @vangheem blind excepts are too broad, what about catching only ``ValueError``?

File: plone/app/querystring/indexmodifiers/query_index_modifiers.py [hunk 2]
```diff
@@ -69,7 +72,10 @@ def __call__(self, value):
                 try:
                     val = datetime.strptime(item, "%Y-%m-%d")
                 except:
-                    val = item
+                    try:
+                        val = datetime.strptime(item, "%m/%d/%Y")
+                    except:
```
Comment (gforcada) [inline plone/app/querystring/indexmodifiers/query_index_modifiers.py]: @vangheem same as above

**Comment by vangheem** [comment]
updated PR, new tests: http://jenkins.plone.org/job/pull-request-5.0/874/

**Pull request merged**